core/pom.xml

```diff
@@ -135,6 +135,12 @@
       <version>${asciidoctorj-pdf.version}</version>
     </dependency>
 
+    <dependency>
+      <groupId>org.yaml</groupId>
+      <artifactId>snakeyaml</artifactId>
+      <version>1.15</version>
+    </dependency>
+
 
     <dependency>
       <groupId>org.jboss.arquillian.junit</groupId>
```

Comment:
Yeah sure. My idea was to have a ConfigHandler interface (filling your config instance) with a method handle(CucumberConfiguration, ExtensionDef). We would then depending the attribute instantiate the right one using Thread.currentThread().getContextClassLoader().loadClass(handlerClass) when it is not a default config. If this loadClass fails we log the dependency to add.

Cant guarantee I can work on it next week but will try to help when Ill get few time.